### ISSUE TITLE
Add ability to rotate the screen up while lying down

### ISSUE THREAD

**Issue by larsbergstrom** [MozillaReality/FirefoxReality#544]
## Hardware
Oculus Go

## Steps to Reproduce
1. Start browser
2. Lie down in bed / recline
3. Recenter screen

## Current Behavior
Everything stays as if you are seated/standing

## Expected Behavior
Able to move the browser or recenter above your face

**Comment by cvan** [comment]
some overlap in #480, but I agree this deserves its own issue. it comes up a lot on Reddit.

**Comment by MortimerGoro** [comment]
I think we only need to detect recenter pose event and apply y position transform based on the headset orientation. I'll give a try :)

**Comment by avrignaud** [comment]
Just note that @thenadj will be having some designs allowing users to
manually set their preferred position here. So want to make sure whatever
automatic shift you explore doesn't impact that 1.1 capability. :)
-- 
Andre Vrignaud — Head of Mixed Reality Platform Strategy | Emerging
Technologies | Mozilla
avrignaud@mozilla.com | +1 (206) 245-6067 <%2B1.206.245.6067> | Skype:
andrevrignaud

**Comment by MortimerGoro** [comment]
@avrignaud yes, it will be compatible, you can still move the windows while lying down

**Comment by MortimerGoro** [comment]
It was a bit more work than expected but PR ready :)

cc @avrignaud @thenadj Some information about the implementation, there are some usability things. I tried to mimic what Oculus Go browser does, so it should be ok.

- The reorientation is triggered using the same recenter controller action.
- The reorientation is set based on where is the user looking at.
- The first orientation when you open the app is always the default one, no matter how the HMD is oriented
- After removing/putting the headset or entering/exiting the reorientation is also reseted to the default one
- I added some thresholds for the roll and yaw angles. That way users can set the default position easily.
